sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_store/feature_store.py

```diff
@@ -0,0 +1,160 @@
+        :param image_build_compute: The name of the compute target to use for building environment
+            Docker images with the container registry is behind a VNet.
+        :type image_build_compute: str
+        :param public_network_access: Whether to allow public endpoint connectivity
+            when a workspace is private link enabled.
+        :type public_network_access: str
+        :param identity: workspace's Managed Identity (user assigned, or system assigned)
+        :type identity: IdentityConfiguration
+        :param primary_user_assigned_identity: The workspace's primary user assigned identity
+        :type primary_user_assigned_identity: str
+        :param kwargs: A dictionary of additional configuration parameters.
+        :type kwargs: dict
+        """
+
+        feature_store_settings = FeatureStoreSettings(
+            compute_runtime=compute_runtime
+            if compute_runtime
+            else ComputeRuntime(spark_runtime_version=DEFAULT_SPARK_RUNTIME_VERSION),
+            offline_store_connection_name=(
+                OFFLINE_STORE_CONNECTION_NAME if materialization_identity and offline_store else None
+            ),
+        )
+        super().__init__(
+            name=name,
+            description=description,
+            tags=tags,
+            kind=FEATURE_STORE_KIND,
+            display_name=display_name,
+            location=location,
+            resource_group=resource_group,
+            hbi_workspace=hbi_workspace,
+            storage_account=storage_account,
+            container_registry=container_registry,
+            key_vault=key_vault,
+            application_insights=application_insights,
+            customer_managed_key=customer_managed_key,
+            image_build_compute=image_build_compute,
+            public_network_access=public_network_access,
+            identity=identity,
+            primary_user_assigned_identity=primary_user_assigned_identity,
+            feature_store_settings=feature_store_settings,
+            **kwargs,
+        )
+        self.offline_store = offline_store
+        self.materialization_identity = materialization_identity
+        self.identity = identity
+
+    @classmethod
+    def _from_rest_object(cls, rest_obj: RestWorkspace) -> "FeatureStore":
+        if not rest_obj:
+            return None
+
+        workspace_object = Workspace._from_rest_object(rest_obj)  # pylint: disable=protected-access
+
+        return FeatureStore(
+            name=workspace_object.name,
+            description=workspace_object.description,
+            tags=workspace_object.tags,
+            compute_runtime=ComputeRuntime._from_rest_object(  # pylint: disable=protected-access
+                workspace_object.feature_store_settings.compute_runtime
+                if workspace_object.feature_store_settings
+                else None
+            ),
+            display_name=workspace_object.display_name,
```

Comment:
How to get offlinestore?

Reply:
From rest object we can not get offline store. I have added the logic in the get api, to call the connection and create offline store